src/hal/components/limit3.comp

```diff
@@ -22,19 +23,24 @@ typedef struct {
 FUNCTION(_) {
     double lin, lout, dt, in_v, min_v, max_v, ramp_a, avg_v, err, dv, dp;
     double min_out, max_out, match_time, est_in, est_out;
+    int inlimit;
 
+    inlimit = 0;
     /* apply first order limit */
     lin = in;
     if ( lin < min_ ) {
 	lin = min_;
+        inlimit = 1;
     }
     if ( lin > max_ ) {
 	lin = max_;
+        inlimit = 1;
     }
 
     if(load) {
 	data.old_in = data.old_out = out = lin;
 	data.old_v = 0;
+        in_limit = 1;
```

Comment:
if (load) is a special case - it stuffs the new values, sets in_limit, and returns without doing any of the rest of the code.
If it set the internal variable inlimit, that value would never be copied to the hal pin.